rascal2/widgets/terminal.py

```diff
@@ -0,0 +1,99 @@
+"""Widget for terminal display."""
+
+import logging
+
+from PyQt6 import QtGui, QtWidgets
+
+from rascal2 import RASCAL2_VERSION
+
+
+class TerminalWidget(QtWidgets.QWidget):
+    """Widget for displaying program output."""
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+
+        self.text_area = QtWidgets.QPlainTextEdit()
+        self.text_area.setReadOnly(True)
+        font = QtGui.QFont()
+        font.setFamily("Courier")
+        font.setStyleHint(font.StyleHint.Monospace)
+        self.text_area.setFont(font)
+        self.text_area.setLineWrapMode(self.text_area.LineWrapMode.NoWrap)
+
+        self.logger = logging.getLogger("log")
+
+        widget_layout = QtWidgets.QVBoxLayout()
+
+        self.scroll_text = QtWidgets.QScrollArea()
+        self.scroll_text.setWidget(self.text_area)
+        self.scroll_text.setWidgetResizable(True)
+        widget_layout.addWidget(self.scroll_text)
```

Comment:
Can you make it so this is scrolled to the left at start?

![Screenshot 2024-09-25 104640](https://github.com/user-attachments/assets/65baaf84-b668-4500-bb24-8f94b0059fd3)

Reply:
i genuinely can't figure out how to do this. it seems the scroll area position gets set once the widget is visible, because some debug printing suggests the scroll area doesn't have a size even by the end of `presenter.create_project`!

Reply:
This works for me
```
        widget_layout = QtWidgets.QVBoxLayout()
        widget_layout.addWidget(self.text_area)

        ...
        ...

        self.write(
            """
 ███████████                       █████████    █████████   █████
░░███░░░░░███                     ███░░░░░███  ███░░░░░███ ░░███
 ░███    ░███   ██████    █████  ███     ░░░  ░███    ░███  ░███
 ░██████████   ░░░░░███  ███░░  ░███          ░███████████  ░███
 ░███░░░░░███   ███████ ░░█████ ░███          ░███░░░░░███  ░███
 ░███    ░███  ███░░███  ░░░░███░░███     ███ ░███    ░███  ░███      █
 █████   █████░░████████ ██████  ░░█████████  █████   █████ ███████████
░░░░░   ░░░░░  ░░░░░░░░ ░░░░░░    ░░░░░░░░░  ░░░░░   ░░░░░ ░░░░░░░░░░░
"""
        )
        self.write_html(f"<b>RasCAL-2:</b> software for neutron reflectivity calculations <b>v{RASCAL2_VERSION}</b>")

        self.text_area.moveCursor(QtGui.QTextCursor.MoveOperation.StartOfLine, QtGui.QTextCursor.MoveMode.MoveAnchor)
````
It turns out the scrollarea was never used because the textEdit creates it own scroll bar thats probably why scroll_text had no size. You can scroll by moving the cursor in the textEdit (see last line)

Reply:
oh, i didn't realise textEdit did that! will remove the scroll area then, thank you